index.js

```diff
@@ -2,32 +2,22 @@
 
 const assert = require('assert')
 const fp = require('fastify-plugin')
-const { initTracer, opentracing, ZipkinB3TextMapCodec } = require('jaeger-client')
+const { initTracerFromEnv, opentracing, ZipkinB3TextMapCodec } = require('jaeger-client')
 const { parse } = require('uri-js');
 const url = require('url');
 
 const { Tags, FORMAT_HTTP_HEADERS } = opentracing
 
 function jaegerPlugin(fastify, opts, next) {
-  assert(opts.serviceName, 'Jaeger Plugin requires serviceName option')
   const { state = {}, initTracerOpts = {}, ...tracerConfig } = opts
   const exposeAPI = opts.exposeAPI !== false
-  const defaultConfig = {
-    sampler: {
-      type: 'const',
-      param: 1
-    },
-    reporter: {
-      logSpans: false
-    }
-  }
 
   const defaultOptions = {
     logger: fastify.log
   }
 
-  const tracer = initTracer(
-    { ...defaultConfig, ...tracerConfig },
+  const tracer = initTracerFromEnv(
```

Comment:
Im wondering if this is good idea to change the implementation so that its not 100% backwards compatible.
My first idea on this task was to have additional parameter to tell plugin if env var should be used or not.. what do you think?

Reply:
Sure, it is good idea. I will do that.